aws_lambda_powertools/metrics/provider/base.py

```diff
@@ -4,16 +4,17 @@
 import logging
 from typing import Any, Callable, Dict, Optional
 
-from typing_extensions import Protocol
+from typing_extensions import Protocol, runtime_checkable
 
 logger = logging.getLogger(__name__)
 
 is_cold_start = True
 
 
+@runtime_checkable
 class MetricsProviderBase(Protocol):
```

Comment:
Let's convert all this as an ABC which requires certain methods, has docstrings, and implements Powertools for AWS Lambda value add (e.g., log_metrics, etc.).

It looked like a good idea to have Protocol when we were to receive a `MetricsProvider` within `Metrics`, but that failed due to the number of specializations for each provider.

We will also not need this anymore, merge with `MetricsBase` and actually call them `BaseProvider(ABC)`.